pkg/webhook/webhook.go

```diff
@@ -832,3 +835,44 @@ func SetInjectHugepageDownApi(hugepageFlag bool) {
 func SetHonorExistingResources(resourcesHonorFlag bool) {
 	honorExistingResources = resourcesHonorFlag
 }
+
+//Watch blocks until either TLS cert & key watcher or HTTP server or termination signal generated
+func Watch(server nri.Service, watcher nri.Service, term chan os.Signal) (err error) {
```

Comment:
can we make this method more generic by making nri.Service as var-args ? 
this would also enable nad cache service for handling termination signal.

Reply:
@pperiyasamy Good idea but I don't know how I could use select to watch all the service signals. If you can think of a pattern let me know.

Reply:
can we try something like this ?

```
	ch := make(chan int)
	for idx := range services {
		go func(serv Service, servIdx int) {
			<-serv.StatusSignal()
			for _, otherService := range services[servIdx+1:] {
				otherService.Quit()
			}
			close(ch)
		}(services[idx], idx)
	}
	<-ch
        return
```